lib/lrmd/lrmd_client.c

```diff
@@ -1981,11 +1981,44 @@ lrmd_api_get_metadata(lrmd_t * lrmd,
     return generic_get_metadata(class, provider, type, output);
 }
 
+static const char*
+lrmd_api_get_version(lrmd_t * lrmd,
+                     const char *class,
+                     const char *provider,
+                     const char *type, enum lrmd_call_options options)
+{
+    const char *version = NULL;
+    char *output = NULL;
+
+    lrmd_api_get_metadata(lrmd, class, provider, type, &output, options);
+
+    if (output) {
+        xmlNode *metadata = string2xml(output);
+        xmlNode *version_xml = get_xpath_object("//resource-agent/@version", metadata, LOG_ERR);
+
+        if (!version_xml) {
+            crm_err("Resource agent version for %s:%s:%s does not specified", class, provider, type);
+        } else {
+            version = crm_element_value(version_xml, "version");
+
+            if (!valid_version_format(version)) {
+                crm_err("Resource agent version for %s:%s:%s has invalid format: %s", class, provider, type, version);
```

Comment:
The OCF specification does not mandate any particular version format, so maybe we should say "unrecognized" rather than "invalid", and give it lower severity, maybe info or notice.